### ISSUE TITLE
Error after run "spago repl"

### ISSUE THREAD

**Issue by Ding-Fan** [purescript/spago#667]
Operating System: macOS Catalina 10.15.5 

After I installed spago, I checked the version

```
$ spago version
0.15.3
```

then run the repl

```
› spago repl
spago: /Users/mrkong/.cache/spago/spago-repl-tmp-1b7495f97ae0a4d7: createDirectory: does not exist (No such file or directory)
```

then I went to `~/.cache/` find out there isn't a directory named "spago", so I `mkdir` it. Try the repl agian, is showed up.

Maybe spago should take care of the directory `~/.cache/spago` ?

**Comment by ozkutuk** [comment]
I experimented a bit to reproduce the issue and here is what I find:

`spago repl` requires a spago project to exist. In the given scenario, since there isn't any local project, it attempts to create a temporary one to launch the REPL on.

✅  If you **have** a local project (which you `init`'ed beforehand), REPL launches successfully (since no need to create a temp. project in cache folder)
✅  If you **don't have** a local project, but **have** `cache/spago`, REPL launches successfully (it can just create the temp. directory without a problem)
❌  If you don't have either, REPL fails

As I was experimenting, I realized `spago init` created `cache/spago` if it doesn't already, whereas `spago repl` by itself did not. Further digging shows that getting the latest package-set tag "asserts" the existence of `cache/spago`. I think we can (and probably should) simply add the same assertion to the `repl` as well (`assertDirectory cacheDir` before attempting to create the temporary directory).

What do you think @f-f?

**Comment by f-f** [comment]
@ozkutuk that's a great diagnosis, well done! :clap: 

I think the least error-prone way to fix this is to just make sure that the directory exists whenever we are in an `Env` that `HasGlobalCache` - so basically taking care of this in the [`RunEnv` module](https://github.com/purescript/spago/blob/master/src/Spago/RunEnv.hs). Makes sense?

**Comment by ozkutuk** [comment]
@f-f Hmm, looks like even the most basic `Env` makes use of global cache (from looking into `withEnv`). So basically we will just make sure global cache dir exists within `withEnv`, right? Or are you thinking of adding something like `withCachedEnv`  variant which holds such logic? There doesn't seem to be anything about `HasGlobalCache` constraint within `RunEnv` module so I was a bit confused 🤔

**Comment by f-f** [comment]
> So basically we will just make sure global cache dir exists within `withEnv`, right?

@ozkutuk that's right! As you can see [here](https://github.com/purescript/spago/blob/010fd37a30b28790e495537bcc4a926f77a5ac83/app/Spago.hs#L36) all the subcommands run inside the `withEnv`, so ensuring that the cache directory exists is best done in there :+1:

**Comment by toraritte** [comment]
May I suggest changing the error message to something that indicate that `spago init` is necessary before the very first run? (And possibly include [@ozkutuk 's awesome summary](https://github.com/purescript/spago/issues/667#issuecomment-660651780))?

Just started to get my feet we on NixOS 20.03 with [`easy-purescript-nix`](https://github.com/justinwoo/easy-purescript-nix), and got the cryptic error message when tried `spago repl`. Did `spago init` which created a `~/.cache/spago`, and the REPL would run after that. (Just a note to myself: deleting `~/.cache/spago` would result in the same error.)

Thanks for considering, and thank you for Spago!

**Comment by f-f** [comment]
@toraritte I usually prefer fixing things upstream rather than recommending workarounds, and in fact @ozkutuk fixed this in #672, so now there should be no need to "init" spago at all, and `spago repl` should just work on the very first run.

I just released [0.16.0](https://github.com/purescript/spago/releases/tag/0.16.0) that includes this fix, let me know if this works for you :slightly_smiling_face:

**Comment by toraritte** [comment]
Thanks, you are right - kind of misunderstood the discussion, plus I was under the impression that I was using the latest version, sorry... Have a good one!

(... and thank you also for inadvertently reminding me to finally find time to read [the latest "Haskell For All" post](http://www.haskellforall.com/2020/07/the-golden-rule-of-software-quality.html) :)
